tests/test_projection_schema_migrations.py

```
@@ -0,0 +1,315 @@
+    SchemaMigration,
+    SchemaMigrationKind,
+    _can_delta_migrate,
+    _plan_schema_upgrade,
+)
+
+# ── policy: every version bump has a migration ─────────────────
+
+
+# Migration registry is required for versions strictly less than
+# the current ``KNOWLEDGE_DB_PROJECTION_SCHEMA_VERSION``.  Versions
+# 1–5 predate the registry (BL-061 introduced the schema-versioning
```

Ruff (0.15.12)

[warning] Comment contains ambiguous `–` (EN DASH). Did you mean `-` (HYPHEN-MINUS)?

(RUF003)